### ISSUE TITLE
Add privacy manifest for Apple

### ISSUE THREAD

**Issue by tony-ditchlabs** [getsentry/sentry-dart#1946]
### Description

See https://github.com/flutter/flutter/issues/131940

Apple announcement here: https://developer.apple.com/support/third-party-SDK-requirements/

May 1st deadline announcement here: https://developer.apple.com/news/?id=3d8a9yyh

**Comment by kahest** [comment]
Hey @tony-ditchlabs thanks for getting in touch! Our sentry-dart SDK version 7.17.0 includes the most recent version of the [privacy manifest file](https://github.com/getsentry/sentry-cocoa/blob/fc015e29d6fbc4039bc6f6286da6147380fa0e72/Sources/Resources/PrivacyInfo.xcprivacy) we ship with our sentry-cocoa SDK. This file includes all privacy APIs we use and the reasons under which we use it.

**Comment by lauroboeni** [comment]
> Hey @tony-ditchlabs thanks for getting in touch! Our sentry-dart SDK version 7.17.0 includes the most recent version of the [privacy manifest file](https://github.com/getsentry/sentry-cocoa/blob/fc015e29d6fbc4039bc6f6286da6147380fa0e72/Sources/Resources/PrivacyInfo.xcprivacy) we ship with our sentry-cocoa SDK. This file includes all privacy APIs we use and the reasons under which we use it.

don't you also need to declare `NSPrivacyAccessedAPICategoryDiskSpace` there?

**Comment by kahest** [comment]
@lauroboeni we removed usage of disk space APIs with [sentry-cocoa 8.21.0](https://github.com/getsentry/sentry-cocoa/releases/tag/8.21.0), therefore we don't declare usage of it.

**Comment by marandaneto** [comment]
Theoretically `sentry_flutter` and `Sentry` are different packages and pods, maybe both require the privacy manifest, but I'm not 100% sure., I've seen packages with both approaches, either having the privacy manifest file only in the Native SDK or both.

**Comment by pawlowskim** [comment]
I will suggest bump up `package_info_plus` minimum version for `sentry_flutter`, otherwise there will be conflict because they have added manifest in version `6.0.0`. Current master should be released ASAP. I know it's over 1.5 month left, but there might be some other unexpected issues with the manifests in general, so preparing for it sooner is better.

**Comment by brustolin** [comment]
> Theoretically sentry_flutter and Sentry are different packages and pods, maybe both require the privacy manifest, but I'm not 100% sure., I've seen packages with both approaches, either having the privacy manifest file only in the Native SDK or both.

AFAIK, technically the app just need to mention a reason once, it does not matter which framework is using the API. 
The app itself could just have one privacy manifest including all the reasons.
If sentry flutter dont use anything else that sentry-cocoa dont use, we dont need to duplicate the privacy manifest.

**Comment by pawlowskim** [comment]
> AFAIK, technically the app just need to mention a reason once, it does not matter which framework is using the API. The app itself could just have one privacy manifest including all the reasons. If sentry flutter dont use anything else that sentry-cocoa dont use, we dont need to duplicate the privacy manifest.

Well, apple also claims each SDK using the specific API needs to define it's own manifest. And without the change you cannot have both libraries in version both have manifest already added.

**Comment by xalikoutis** [comment]
Please bump up `package_info_plus` . It is a stopper to update other packages that fix privacy manifest.
26 days more..... and counting down

**Comment by pawlowskim** [comment]
@xalikoutis I think the safest way to go forward with sentry right now is to use specific commit, where it's already updated:
```yaml
    sentry_flutter:
        git:
            url: 'https://github.com/getsentry/sentry-dart'
            path: 'flutter'
            ref: 5152a56dc09336dbdb96a3058feadeba3e3bb02c
```

**Comment by ueman** [comment]
```yaml
dependency_overrides:
  package_info_plus: ^7.0.0
```

Is another easy solution

**Comment by denrase** [comment]
@buenaflor In order to bump `package_info_plus`, we'd also need to bump `sdk` and `flutter`:

```yaml
environment:
  sdk: ">=3.3.0 <4.0.0"
  flutter: ">=3.19.0"
```

Should we document @ueman's solution for users experiencing this issue?

**Comment by denrase** [comment]
@kahest @buenaflor @stefanosiano Correct me if I'm wrong, but since we have multiple packages which use other dependencies (isar, drift), we'd need to check if all of them provide correct privacy manifests and bump their min versions to where it's included. Or at least document min version overrides, as this is "only" affecting iOS users and others might prefer to keep compatibility with lower versions. WDYT?

**Comment by buenaflor** [comment]
> we'd need to check if all of them provide correct privacy manifests and bump their min versions to where it's included

I believe so. I would assume that most people using flutter also have an ios app but yeah, not sure if we should just bump the min version or document overrides. If the majority of our users also have an ios app then I'd say we can bump it.

thoughts?

**Comment by kahest** [comment]
I vote for documenting min version overrides

**Comment by himanshu-newstok** [comment]
I've been closely following Apple's recent announcement mandating the inclusion of PrivacyInfo.xcprivacy in third-party SDK packages for iOS integration. As part of my investigation, I have thoroughly reviewed the Sentry Flutter repository, particularly the iOS folder, but I couldn't locate the PrivacyInfo.xcprivacy file.

Could someone from the Sentry team kindly confirm whether PrivacyInfo.xcprivacy has been included in the Sentry Flutter package for iOS integration? If it's not currently included, are there plans to add it in future updates?

Your prompt response to this inquiry would be greatly appreciated.

**Comment by buenaflor** [comment]
hey @himanshu-newstok it is included starting with sentry flutter version `7.17.0` from our sentry cocoa sdk, read more [here](https://github.com/getsentry/sentry-dart?tab=readme-ov-file#apple-privacy-manifest)
